lib/node_modules/@stdlib/math/base/special/lnf/lib/main.js

```diff
@@ -0,0 +1,172 @@
+*
+* ```text
+* Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var toWordf = require( '@stdlib/number/float32/base/to-word' );
+var fromWordf = require( '@stdlib/number/float32/base/from-word' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var NINF = require( '@stdlib/constants/float32/ninf' );
+var FLOAT32_EXPONENT_MASK = require( '@stdlib/constants/float32/exponent-mask' );
+var FLOAT32_EXPONENT_BIAS = require( '@stdlib/constants/float32/exponent-bias' );
+var FLOAT32_SIGNIFICAND_MASK = require( '@stdlib/constants/float32/significand-mask' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var polyvalP = require( './polyval_p.js' );
+var polyvalQ = require( './polyval_q.js' );
+
+
+// VARIABLES //
+
+var LN2_HI = 6.9313812256e-01; // 0x3f317180
+var LN2_LO = 9.0580006145e-06; // 0x3717f7d1
+var TWO25 = 3.355443200e+07;   // 0x4c000000
+var ONE_THIRD = 0.33333333333333333;
+
+
+// MAIN //
+
+/**
+* Evaluates the natural logarithm of a single-precision floating-point number.
+*
+* @param {NonNegativeNumber} x - input value
+* @returns {number} function value
+*
+* @example
+* var v = lnf( 4.0 );
+* // returns ~1.386
+*
+* @example
+* var v = lnf( 0.0 );
+* // returns -Infinity
+*
+* @example
+* var v = lnf( Infinity );
+* // returns Infinity
+*
+* @example
+* var v = lnf( NaN );
+* // returns NaN
+*
+* @example
+* var v = lnf( -4.0 );
+* // returns NaN
+*/
+function lnf( x ) {
+	var hfsq;
+	var ix;
+	var t2;
+	var t1;
+	var dk;
+	var k;
+	var R;
+	var f;
+	var i;
+	var j;
+	var s;
+	var w;
+	var z;
+
+	if ( x === 0.0 ) {
+		return NINF;
+	}
+	if ( isnan( x ) || x < 0.0 ) {
+		return NaN;
+	}
+	x = float64ToFloat32( x );
+	ix = toWordf( x );
+	k = 0;
+
+	// x < 2**-126
+	if ( ix < 0x00800000 ) {
+		k = float64ToFloat32( k - 25 );
+
+		// Subnormal number, scale up x
+		x = float64ToFloat32( x * TWO25 );
+		ix = toWordf( x );
+	}
+	if ( ix >= FLOAT32_EXPONENT_MASK ) {
+		return float64ToFloat32( x + x );
+	}
+	k = float64ToFloat32( k + float64ToFloat32( ( ix >> 23 ) - FLOAT32_EXPONENT_BIAS ) ); // eslint-disable-line max-len
+	ix &= FLOAT32_SIGNIFICAND_MASK;
+	i = float64ToFloat32( ix + ( 0x95f64 << 3 ) ) & 0x800000;
+
+	// Normalize x or x/2
+	x = fromWordf( ix | ( i ^ 0x3f800000 ) );
+	k = float64ToFloat32( k + ( i >> 23 ) );
+	f = float64ToFloat32( x - 1.0 );
+
+	// -2**-9 <= f < 2**-9
+	if ( ( FLOAT32_SIGNIFICAND_MASK & float64ToFloat32( 0x8000 + ix ) ) < 0xc000 ) { // eslint-disable-line max-len
+		if ( f === 0.0 ) {
+			if ( k === 0 ) {
+				return 0.0;
+			}
+			dk = k;
```

Comment:
Here, int32 k is being cast to double in the C implementation. We prob don't need this assignment in the JS version.